### ISSUE TITLE
Do not use Count/Length in Assert

### ISSUE THREAD

**Issue by RalfKoban** [RalfKoban/MiKo-Analyzers#436]
Some developers tend to verify collections via something like
```c#
Assert.AreEqual(0, collection.Count);

Assert.That(collection.Count, Is.EqualTo(0));

Assert.That(collection.Count, Is.Zero);
```

This makes it hard to inspect what's contained in the collection if the count is not equal to zero. Instead, something like
```c#
Assert.That(collection, Is.Empty);
```
makes it much easier to find out what's contained when it fails.

So we should warn about the usage of `.Count`, `.Length` or the Linq call `.Count()`.
